### ISSUE TITLE
Base Models missing an `id` field in REST API

### ISSUE THREAD

**Issue by fussybeaver** [PagerDuty/api-schema#77]
I'm trying to use the openapi schema to generate a client API for the REST API, but noticed during an integration test that none of the base models (e.g. Service, Incident, EscalationPolicy etc.) have an `id` field. 

https://github.com/PagerDuty/api-schema/blob/master/reference/REST/openapiv3.json

So, for example - Service:
https://github.com/PagerDuty/api-schema/blob/master/reference/REST/openapiv3.json#L2385-L2495

If you do a `GET` request to the `/services/...` endpoint(s), you will get this JSON structure:

```
{
  "services": [
    {
      "id": "PHL4Y2Z",             <---- this field is missing from the openapi schema
      "name": "Foo",
      "description": null,
      "created_at": "2020-07-12T13:33:30+01:00",
      "updated_at": "2020-07-12T13:33:30+01:00",
      "status": "active",
      "teams": [
        {
          "id": "PQMRJBY",
          "type": "team_reference",
          "summary": "Platform Engineering",
          "self": "https://api.pagerduty.com/teams/PQMRJBY",
          "html_url": "https://example-example-uk.pagerduty.com/teams/PQMRJBY"
        }
      ],
      "alert_creation": "create_incidents",
      "addons": [],
      "scheduled_actions": [],
      "support_hours": null,
      "last_incident_timestamp": null,
      "escalation_policy": {
        "id": "PWZCAZT",
        "type": "escalation_policy_reference",
        "summary": "rsk",
        "self": "https://api.pagerduty.com/escalation_policies/PWZCAZT",
        "html_url": "https://example-example-uk.pagerduty.com/escalation_policies/PWZCAZT"
      },
      "incident_urgency_rule": {
        "type": "constant",
        "urgency": "high"
      },
      "acknowledgement_timeout": null,
      "auto_resolve_timeout": null,
      "alert_grouping": null,
      "alert_grouping_timeout": null,
      "integrations": [],
      "response_play": null,
      "type": "service",
      "summary": "Foo",
      "self": "https://api.pagerduty.com/services/PHL4Y2Z",
      "html_url": "https://example-example-uk.pagerduty.com/service-directory/PHL4Y2Z"
    }
  ],
  "limit": 25,
  "offset": 0,
  "total": null,
  "more": false
}
```

**Comment by bzmw** [comment]
Hey! Thanks for the report. Looking into fixing this now.

**Comment by bzmw** [comment]
That should fix it @fussybeaver ! 

Thanks for the report, if you find any other inconsistencies please let me know. I'd also love to know what you are using your autogenerated client for, and what language you are generating it in? I'm assuming you're using openapi-generator.

**Comment by fussybeaver** [comment]
Thanks for the fix. I'm using the openapi-generator for a rust api client, which I will probably open source if it gets mature enough. Unfortunately I think the new changes need some adjustments to make them work well within the scope of generating API's.

I managed to get the models generating the additional fields, but only after modifying the pagerduty swagger json. After reading the specification, I think this is mainly due to limitations in the openapi generator though...

After some debugging, it looks like the default openapi code generator doesn't resolve `$ref` beyond first class types when creating a model for composite types, so when one does:

```
"Service": {
        "allOf": [
          {
            "$ref": "#/components/schemas/Tag/allOf/0"
          },
          { ... } ] }
```

This won't resolve into the correct polymorphic subtype of `Tag`. 

I changed all these references to "#/components/schemas/Tag", which got me a little further. However, it also looks like the default code generator only builds composite types from one complex type and other `$ref` types. 

So, when one does:
```
"Tag": {
        "allOf": [
          {
            "type": "object",
            "properties": { ... }
          },
          {
            "type": "object",
            "properties": { ... }
          } ] }
```

This will fail to build all properties, preferring to use the last type. 

I think these problems are mainly because polymorphism in the openapi is a little bit naive. 

The schema I ended up with:
https://gist.github.com/fussybeaver/4ba9a98fed7c0c9c836a04ade3a108f3
